src/client.rs

```diff
@@ -11,14 +19,32 @@ impl LedgerClient {
         ledger_derivation_path: Option<usize>,
         chain_id: u64,
         rpc_url: String,
-    ) -> anyhow::Result<Self> {
+    ) -> Result<Self, LedgerClientError> {
         let wallet = Ledger::new(
             HDPath::LedgerLive(ledger_derivation_path.unwrap_or(0)),
             chain_id,
         )
-        .await?;
-        let provider = Provider::<Http>::try_from(rpc_url.clone())?;
-        let client = SignerMiddleware::new_with_provider_chain(provider, wallet).await?;
+        .await
+        .map_err(|err| {
+            LedgerClientError::CreateLedgerClientError(format!(
+                "Failed to instantiate Ledger device: {}",
+                err.to_string()
+            ))
+        })?;
+        let provider = Provider::<Http>::try_from(rpc_url.clone()).map_err(|err| {
+            LedgerClientError::CreateLedgerClientError(format!(
+                "Failed to instantiate provider: {}",
```

Comment:
maybe this should be a distinct error enum indicating its a "rpc provider setup error", rather than just making the string different?